endpoints/get_address_transactions.py

```diff
@@ -60,3 +63,42 @@ async def get_transactions_for_address(
     return {
         "transactions": tx_list
     }
+
+@app.get("/addresses/{kaspaAddress}/transactions/v2",
+         response_model=List[TxModel],
+         response_model_exclude_unset=True,
+         tags=["Kaspa addresses transactions"])
+async def get_transactions_for_address_v2(
+        kaspaAddress: str = Path(
+            description="Kaspa address as string e.g. "
+                        "kaspa:pzhh76qc82wzduvsrd9xh4zde9qhp0xc8rl7qu2mvl2e42uvdqt75zrcgpm00",
+            regex="^kaspa\:[a-z0-9]{61}$"),
+        limit: int = Query(
+            description="The number of records to get",
+            ge=1,
+            le=500,
+            default=50),
+        offset: int = Query(
+            description="The offset from which to get records",
+            ge=0,
+            default=0),
+        fields: str = "",
+    ):
+    """
+    Get all transactions for a given address from database.
+    And then get their related full transactiond data
+    """
+
+    async with async_session() as s:
+        # Doing it this way as opposed to adding it directly in the IN clause
+        # so I can re-use the same result in tx_list, TxInput and TxOutput
+        txWithinLimitOffset = await s.execute(select(TxAddrMapping.transaction_id)
+            .filter(TxAddrMapping.address == kaspaAddress)
+            .limit(limit)
+            .offset(offset)
+            .order_by(TxAddrMapping.block_time.desc())
```

Comment:
Let's query/join the 'transactions' here I guess without using the search function. Will be fast - no?

Reply:
SELECT * FROM 
  transactions,
(SELECT * FROM tx_id_address_mapping WHERE address='kaspa:-....'
ORDER by block_time DESC
LIMIT 50
OFFSET 10000) as subquery

WHERE transactions.transaction_id = subquery.transaction_id

Reply:
I used the function that exists in search because that one actually performs 3 queries: one for the TxModel and, depending on if `fields` contains `inputs` or `outputs` it will query those too.

If we only did the one query in `search` then I would agree that we would simply do the subquery like this.

However, if we don't really need the tx_output query and tx_input query that was being done in `search` I can make this endpoint do the query itself.